addons/csw/functions/fnc_assemble_deployTripod.sqf

```diff
@@ -0,0 +1,66 @@
+#include "script_component.hpp"
+/*
+ * Author: TCVM
+ * Deploys the tripod
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player] call ace_csw_fnc_assemble_deployTripod
+ *
+ * Public: No
+ */
+
+[{
+    params ["_player"];
+    TRACE_1("assemble_deployTripod",_player);
+
+    // Remove the tripod from the launcher slot
+    private _secondaryWeaponClassname = secondaryWeapon _player;
+    _player removeWeaponGlobal (secondaryWeapon _player);
+
+    private _onFinish = {
+        params ["_args"];
+        _args params ["_player", "_secondaryWeaponClassname"];
+        TRACE_2("deployTripod finish",_player,_secondaryWeaponClassname);
+
+        private _tripodClassname = getText(configFile >> "CfgWeapons" >> _secondaryWeaponClassname >> QUOTE(ADDON) >> "deploy");
+
+        // Create a tripod
+        private _cswTripod = createVehicle [_tripodClassname, [0, 0, 0], [], 0, "NONE"];
+
+        private _posATL = _player getRelPos [2, 0];
+        _posATL set [2, ((getPosATL _player) select 2) + 0.5];
+
+        _cswTripod setDir (direction _player);
```

Comment:
Some small tripods can suffer from rollover even on slight slope.
```suggestion
        _cswTripod setDir (direction _player);
        _cswTripod setCenterOfMass ((getCenterOfMass _cswTripod) vectorAdd [0,0,-0.3]);
```

Reply:
Should we really fix a bug in the 3D model of some tripods?

Reply:
ace still does this in ace_vehicles
```
class Extended_Init_EventHandlers {
    class StaticWeapon {
        class ACE_FixMass {
            init = QUOTE(if (local (_this select 0)) then {(_this select 0) setMass (getMass (_this select 0) max 250)};);
```
so for full static weapons's we try to fix them, so it wouldn't be out of line to fix the tripod models as well
we control most of the p3ds for tripods, so we might be able to fix by moving mass
and perhaps this command would work better than setMass for the full weapon?

Reply:
FYI setting center of mass has one issue, if you can pack the weapon into the backpacks it will reset after it is assembled again.

Reply:
> Should we really fix a bug in the 3D model of some tripods?
This happens on small and narrow third party models. RHS/CUP SPG and Metis have very narrow models. So I don't think that it can be/will be fixed.

> FYI setting center of mass has one issue, if you can pack the weapon into the backpacks it will reset after it is assembled again.
That is why it should be done on each createVehicle. Not shure about Drag/Carry actions after placement.

Reply:
> That is why it should be done on each createVehicle. 

Weapon assembly/disassembly does not fire `createVehicle`, the weapon is hidden at some coordinates until assembled again.

Reply:
@veteran29 In the way that CSW works `createVehicle` does fire